File: src/main/java/com/arnas/klatrebackend/features/routes/RouteRepository.java
package com.arnas.klatrebackend.features.routes;

import org.springframework.lang.NonNull;

import java.util.Optional;
import java.util.List;

public interface RouteRepository {

    Optional<Route> getRouteById(long routeId);
    List<Route> getRoutesByPlace(long placeId, int page, int limit, boolean pagingEnabled);
    List<Route> getRoutesByPlace(long placeId);

    long addRoute(@NonNull RouteDTO routeDTO, String imageId, long userId);
    int updateRoute(@NonNull Route route);
    int deleteRoute(long routeId);
    int getNumRoutesInPlace(long placeId, boolean countActive);

}

File: src/main/kotlin/com/arnas/klatrebackend/features/routes/RouteRepository.kt
package com.arnas.klatrebackend.features.routes

import java.util.Optional

interface RouteRepository {
    fun getRouteById(routeId: Long): Optional<Route>
    fun getRoutesByPlace(placeId: Long, page: Int, limit: Int, pagingEnabled: Boolean): List<Route>
    fun getRoutesByPlace(placeId: Long): List<Route>
    fun addRoute(routeDTO: RouteDTO, imageId: String?, userId: Long): Long
    fun updateRoute(route: Route): Int
    fun deleteRoute(routeId: Long): Int
    fun getNumRoutesInPlace(placeId: Long, countActive: Boolean): Int
}

